index.js

```diff
@@ -387,17 +390,48 @@
     function(x) { return $.ValidNumber.test(x) && isFinite(x); }
   );
 
+  //  Integer :: Type
+  $.Integer = $.NullaryType(
+    'sanctuary-def/Integer',
+    function(x) {
+      return $.ValidNumber.test(x) &&
+             Math.floor(x) === x &&
+             x >= MIN_SAFE_INTEGER &&
+             x <= MAX_SAFE_INTEGER;
+    }
+  );
+
+  /*jshint -W041, -W116*/
```

Comment:
I'm happy to remove these rules entirely. I have faith in our ability to flag inappropriate use of `==`/`!=` during reviews.